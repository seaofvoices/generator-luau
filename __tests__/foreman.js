import { expect, it, describe, jest, beforeEach } from '@jest/globals'
import { getFileContent, setupGenerator, usePolly } from './utils'

const projectName = 'new-project'
let spawnMock = jest.fn()
let answers = {}

const setupContext = async () =>
  await setupGenerator('foreman')
    .withArguments([projectName])
    .withAnswers(answers)
    .withSpawnMock(spawnMock)

beforeEach(() => {
  spawnMock = jest.fn()
  answers = {
    tools: [],
  }
})

describe('default tools', () => {
  beforeEach(() => {
    usePolly('default-tools-requests')
  })

  it('creates a foreman.toml', async () => {
    const context = await setupContext()

    expect(getFileContent(context, 'foreman.toml')).toMatchSnapshot()
  })

  it('runs the foreman install command', async () => {
    await setupContext()

    expect(spawnMock).toHaveBeenCalledWith(
      'spawnSync',
      'foreman',
      ['install'],
      expect.any(Object)
    )
  })
})

describe('all tools', () => {
  beforeEach(async () => {
    answers.tools = [
      'selene',
      'stylua',
      'luau-lsp',
      'darklua',
      'rojo',
      'tarmac',
      'run-in-roblox',
      'lune',
      'mantle',
    ]

    usePolly('all-tools-requests')
  })

  it('creates a foreman.toml', async () => {
    const context = await setupContext()

    expect(getFileContent(context, 'foreman.toml')).toMatchSnapshot()
  })

  it('runs the foreman install command', async () => {
    await setupContext()

    expect(spawnMock).toHaveBeenCalledWith(
      'spawnSync',
      'foreman',
      ['install'],
      expect.any(Object)
    )
  })

  it('runs the lune setup command', async () => {
    await setupContext()

    expect(spawnMock).toHaveBeenCalledWith(
      'spawnSync',
      'lune',
      ['setup'],
      expect.any(Object)
    )
  })
})
