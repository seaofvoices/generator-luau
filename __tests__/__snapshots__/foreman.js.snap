// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all tools creates a foreman.toml 1`] = `
"[tools]
darklua = { github = "seaofvoices/darklua", version = "=0.13.1"}
luau-lsp = { github = "JohnnyMorganz/luau-lsp", version = "=1.32.3"}
lune = { github = "lune-org/lune", version = "=0.8.8"}
mantle = { github = "blake-mealey/mantle", version = "=0.11.16"}
rojo = { github = "rojo-rbx/rojo", version = "=7.4.4"}
run-in-roblox = { github = "rojo-rbx/run-in-roblox", version = "=0.3.0"}
selene = { github = "Kampfkarren/selene", version = "=0.27.1"}
stylua = { github = "JohnnyMorganz/StyLua", version = "=0.20.0"}
tarmac = { github = "rojo-rbx/tarmac", version = "=0.7.5"}
"
`;

exports[`default tools creates a foreman.toml 1`] = `
"[tools]
darklua = { github = "seaofvoices/darklua", version = "=0.13.1"}
luau-lsp = { github = "JohnnyMorganz/luau-lsp", version = "=1.32.3"}
selene = { github = "Kampfkarren/selene", version = "=0.27.1"}
stylua = { github = "JohnnyMorganz/StyLua", version = "=0.20.0"}
"
`;
